### PR TITLE
fix 'banned user doesn't disappear from pal until refresh'

### PR DESCRIPTION
**Pull request by howard-stearns**
fixes https://highfidelity.fogbugz.com/f/cases/3362/banned-user-doesn-t-disappear-from-pal-until-refresh

### CODE REVIEW

File: interface/resources/qml/hifi/Pal.qml [hunk 1]
```diff
@@ -360,10 +360,8 @@ Rectangle {
                     Users[styleData.role](model.sessionId)
                     UserActivityLogger["palAction"](styleData.role, model.sessionId)
                     if (styleData.role === "kick") {
-                        // Just for now, while we cannot undo "Ban":
-                        userModel.remove(model.userIndex)
-                        delete userModelData[model.userIndex] // Defensive programming
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/Pal.qml]: don't delete from an array, as it clears the element without removing it

File: interface/resources/qml/hifi/Pal.qml [hunk 2]
```diff
@@ -360,10 +360,8 @@ Rectangle {
...
-                        sortModel()
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/Pal.qml]: we're removing the ModelList item directly, so no need to sort (and rebuild)

**Comment by howard-stearns** [comment]
test plan: requires two users and a sandbox you own.
- Open pal, and ban the other guy.
- He should disappear from pal.
- try refresh or sort, and the remaining users should work as expected.

(Don't forget to go into settings and unban your friend.)

**Comment by hifi-gustavo** [comment]
Build console output for commit: a715a694f0d2ffb445e3feb0438b5f2e4af3cc98 available at: https://gist.github.com/c4185fabb67ee3ee4c19e0c5ccb99a3e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9805-a715a69.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9805-a715a69.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 10/Ubuntu 16.04
QA Pass

Compared to current master, this PR works as outlined.  General smoke test pass, PAL works otherwise as expected.

File: interface/resources/qml/hifi/Pal.qml [hunk 3]
```diff
@@ -360,10 +360,8 @@ Rectangle {
...
+                        userModelData.splice(model.userIndex, 1)
```
Comment (ZappoMan) [inline interface/resources/qml/hifi/Pal.qml]: coding standard: needs trailing `;`

File: interface/resources/qml/hifi/Pal.qml [hunk 4]
```diff
@@ -360,10 +360,8 @@ Rectangle {
...
+                        userModel.remove(model.userIndex) // after changing userModelData, b/c ListModel can frob the data
```
Comment (ZappoMan) [inline interface/resources/qml/hifi/Pal.qml]: coding standard: needs trailing `;`

**Comment by ZappoMan** [comment]
note: @howard-stearns - I see in some of the non-changed code there are also missing trailing `;`... please fix those too.

**Comment by howard-stearns** [comment]
I also agree that .qml should follow our standards for .js.
However, an earlier sweep through this file explicitly converted line endings to not use semicolons, for readability. I chose not to block it at that time.

I have no objection to changing it all back, if we are in agreement to do so going forward.

**Comment by howard-stearns** [comment]
ok, i've add the semicolons throughout, and fixed the whitespace (indentation) as well.

**Comment by howard-stearns** [comment]
@ZappoMan with the requested change made, can you remove your block? Because the requested change affects every line of the file, this interferes with all other work.

**Comment by ZappoMan** [comment]
@howard-stearns - approved - but it looks like there is a merge conflict.

**Comment by howard-stearns** [comment]
something went wrong with the merge. dnm until I straighten it out.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8a489f36f41143071ad1abab37ae4ac5d9b86ef9 available at: https://gist.github.com/c4185fabb67ee3ee4c19e0c5ccb99a3e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9805-8a489f3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9805-8a489f3.dmg

**Comment by howard-stearns** [comment]
fixed broken merge and retested. just waiting for build

**Comment by hifi-gustavo** [comment]
Build console output for commit: cb41b9135bd7feca8f984bf7e48354030641d38b available at: https://gist.github.com/c4185fabb67ee3ee4c19e0c5ccb99a3e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9805-cb41b91.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9805-cb41b91.dmg

**Pull request merged**